src/robotFactory.js

```diff
@@ -1,10 +1,115 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    if ('x' in coords && 'y' in coords) {
+      this.coords = coords;
+    } else if ('x' in coords) {
+      this.coords = {
+        x: coords.x,
+        y: 0,
+      };
+    } else if ('y' in coords) {
+      this.coords = {
+        x: 0,
+        y: coords.y,
+      };
+    } else {
+      this.coords = {
+        x: 0,
+        y: 0,
+      };
+    }
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    if (step >= 0) {
+      this.coords.y += step;
+    }
+
+    return this;
+  }
+
+  goBack(step = 1) {
+    if (step >= 0) {
+      this.coords.y -= step;
+    }
+
+    return this;
+  }
+
+  goRight(step = 1) {
+    if (step >= 0) {
+      this.coords.x += step;
+    }
+
+    return this;
+  }
+
+  goLeft(step = 1) {
+    if (step > 0) {
+      this.coords.x -= step;
+    }
+
+    return this;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${
+      this.chipVersion}, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
```

Comment:
let's make a code cleaner
```suggestion
  constructor(
   name,
   weight,
   coords,
   chipVersion
  ) {
```